### PR TITLE
Add seed to nested sampling

### PR DESCRIPTION
**Pull request by fnovak42**
As discussed in issue #573, a seed was added to the rng used by the `DynamicNestedSampler`

### CODE REVIEW

**Review by dvandyk — CHANGES_REQUESTED** [review]
This looks reasonable, thanks! However, the change should also be reflected in `eos-analysis`, where the same default value for the seed should be used as in the task.

**Comment by fnovak42** [comment]
For simplicity, I've added just the `int` option in the task and parser specifications for the seed. Technically, we could also use an array or `None`, but I'm not sure if this is practically relevant.

File: src/scripts/eos-analysis [hunk 1]
```diff
@@ -249,6 +249,10 @@ The output file will be stored in EOS_BASE_DIRECTORY/POSTERIOR/nested.
         help = 'The maximum number of iterations. Iterations may stop earlier if the termination condition is reached.',
         dest = 'maxiter', action = 'store', type = int, default = None
     )
+    parser_sample_nested.add_argument('-s', '--rng-seed',
```
Comment (mreboud) [inline src/scripts/eos-analysis:252]: Can you use the command `--use-random-seed` for consistency with the `find-mode` task?

**Review by mreboud — CHANGES_REQUESTED** [review]
Have you been able to check that it works as expected? I would run 3 nested samplings: 2 with the default seed and a with another one. Then you can check that the events in the two first samplings are the same and don't match the ones in the third sampling.

**Comment by fnovak42** [comment]
> Have you been able to check that it works as expected? I would run 3 nested samplings: 2 with the default seed and a with another one. Then you can check that the events in the two first samplings are the same and don't match the ones in the third sampling.

All right, that sound like a better way to check it, I will try that. I just ran two samplings, one with the default seed and one with a different one.

**Review by dvandyk — APPROVED** [review]
Looks good now, thank you @fnovak42!

**Pull request merged**